rustls-cert-gen/src/cert/mod.rs

```diff
@@ -0,0 +1,44 @@
+use rcgen::{Certificate, CertificateParams, Error};
+use std::{fs::File, io, path::Path};
+
+mod params;
+pub use params::BuildParams;
+
+mod ca;
+mod entity;
+mod signature;
+
+pub trait CertParams {
+	fn params(&self) -> &CertificateParams;
+}
```

Comment:
This feels like another trait that I'm not sure is necessary: it's probably clearer to have `Ca` and `EndEntity` define their own `params()` fn. Like the `Cert` trait there isn't anywhere in the existing `main.rs` where you want to treat both EE and CAs uniformly without knowing which type you have in-hand.